bokehjs/test/integration/annotations/legend.ts

```diff
@@ -438,4 +438,56 @@ describe("Legend annotation", () => {
 
     await display(p)
   })
+
+  describe("should support as border_line_dash pattern value", () => {
+    const p = fig([200, 200])
+    const x = [1, 2, 3, 4, 5]
+    const y1 = [2, 3, 4, 5, 6]
+    const y2 = [3, 4, 5, 6, 7]
+
+    p.line(x, y1, {legend_label: "Temp.", line_color: "blue"})
+    p.line(x, y2, {legend_label: "Objects", line_color: "red"})
+
+    p.legend.location = "top_left"
+    p.legend.border_line_color = "black"
+
+    it("default", async () => {
+      await show_with_exported(p)
+    })
+
+    it("solid", async () => {
+      p.legend.border_line_dash = "solid"
+      await show_with_exported(p)
+    })
+
+    it("dotdash", async () => {
+      p.legend.border_line_dash = "dotdash"
+      await show_with_exported(p)
+    })
+
+    it("dashdot", async () => {
+      p.legend.border_line_dash = "dashdot"
+      await show_with_exported(p)
+    })
+
+    it("dashed", async () => {
+      p.legend.border_line_dash = "dashed"
+      await show_with_exported(p)
+    })
+
+    it("dotted", async () => {
+      p.legend.border_line_dash = "dotted"
+      await show_with_exported(p)
+    })
+
+    it("Custom 0", async () => {
+      p.legend.border_line_dash = [2, 4, 3, 4]
+      await show_with_exported(p)
+    })
+
+    it("Custom 1", async () => {
+      p.legend.border_line_dash = [2, 4, 3, 4, 4, 4, 5, 4, 6, 4, 7, 4, 8, 4, 9, 4, 10, 4]
+      await show_with_exported(p)
+    })
```

Comment:
I would prefer to have these patterns in test description, e.g. `it("custom pattern 2, 4, 3, 4 (even number of items)"). I would shorten the second pattern while keeping small and large values and make it an odd number of items.